src/NuGet.Clients/NuGet.VisualStudio.Common/Etw/NuGetExtensibilityEtw.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+
+namespace NuGet.VisualStudio.Etw
+{
+    public static class NuGetExtensibilityEtw
+    {
+        public static EventSource EventSource { get; } = new EventSource("NuGet-VS-Extensibility");
+
+        public static EventSourceOptions StartEventOptions { get; } =
+            new EventSourceOptions()
+            {
+                Opcode = EventOpcode.Start,
+                Level = EventLevel.Informational,
+                ActivityOptions = EventActivityOptions.Detachable
```

Comment:
How this "EventActivityOptions.Detachable" used later? Do we use for some correlation?

Reply:
[The docs for EventActivityOptions](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventactivityoptions?view=netframework-4.8#system-diagnostics-tracing-eventactivityoptions-detachable) say this for `Detachable`:

> Allow overlapping activities. By default, activity starts and stops must be property nested. That is, a sequence of Start A, Start B, Stop A, Stop B is not allowed will result in B stopping at the same time as A.

Since there's nothing preventing different VS components, or even the same VS component, to call any of our APIs in parallel, we can certainly be in the "start A, start B, stop A, stop B" scenario. If you take a trace of devenv with these changes, it's not uncommon to see parallel project nominations and GetInstalledAsync calls.